x-pack/plugins/apm/common/critical_path/get_critical_path.test.ts

```diff
@@ -12,17 +12,18 @@ import { getCriticalPath } from './get_critical_path';
 
 describe('getCriticalPath', () => {
   function getCriticalPathFromEvents(events: ApmFields[]) {
-    const waterfall = getWaterfall(
-      {
+    const entryTransaction = dedot(events[0]!, {}) as Transaction;
```

Comment:
Not directly related to this PR:
I looked at `dedot` and I'm still not 100% what it does. It doesn't have any tests afaict. 
Does it turn `foo.bar.baz` into `{foo: {bar: 'baz'}}` ?

We should remove the second argument from `dedot` and return the value instead. Afaict we don't use the second arg anywhere. 

```suggestion
    const entryTransaction = dedot(events[0]!) as Transaction;
```

Reply:
The second argument is there so we are not required to create a new object every time.

Reply:
But it's never needed from the current usage afaict.

Reply:
... so if anything it seems like an edge case and should not be required.